invenio_app_ils/records/serializers/item.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2018 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Ils serializers."""
+
+from flask import has_request_context
+from invenio_records_rest.serializers.json import JSONSerializer
+
+from invenio_app_ils.permissions import loan_owner_permission
+
+
+class ItemJSONSerializer(JSONSerializer):
+    """Serialize and filter item circulation status."""
+
+    FILTER_KEYS = [
+        "loan_pid",
+        "patron_pid",
+        "start_date",
+        "end_date",
+        "extension_count",
+        "request_expire_date",
+    ]
+
+    def transform_record(self, pid, record, links_factory=None, **kwargs):
+        """Transform record into an intermediate representation."""
+        item = super(ItemJSONSerializer, self).transform_record(
+            pid,
+            record,
+            links_factory=links_factory,
+            **kwargs
+        )
+        self.filter_circulation_status(item)
+        return item
+
+    def transform_search_hit(self, pid, record_hit, links_factory=None,
+                             **kwargs):
+        """Transform search result hit into an intermediate representation."""
+        hit = super(ItemJSONSerializer, self).transform_search_hit(
+            pid,
+            record_hit,
+            links_factory=links_factory,
+            **kwargs
+        )
+        self.filter_circulation_status(hit)
+        return hit
+
+    def filter_circulation_status(self, data):
+        """Filter circulation status depending on user permissions."""
+        if "circulation_status" in data["metadata"]:
+            circulation_status = data["metadata"]["circulation_status"]
+            patron_pid = circulation_status.get("patron_pid", None)
+            if not patron_pid:
+                return circulation_status
+
+            allowed = True
+            if has_request_context():
```

Comment:
I would say that you don't need `has_request_context`, you should always be in a request context, no?

Reply:
This was a fix for the cli commands as I remember :)

Reply:
@ntarocco Yes, it looks like you're right. I removed it and no issues. I removed the check since I'm not sure in what situation that can actually happen.